lib/Helper/ParameterBagHelper.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This source file is available under the terms of the
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+namespace Pimcore\Helper;
+
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+/**
+ * Helper class for safely extracting typed values from ParameterBag instances.
+ *
+ * This helper addresses Symfony 7 breaking changes where ParameterBag::getInt()
+ * and ParameterBag::getBool() now throw UnexpectedValueException instead of
+ * returning fallback values. This helper uses the filter() method internally
+ * with FILTER_NULL_ON_FAILURE to maintain backward compatibility.
+ *
+ * @internal
+ */
+final class ParameterBagHelper
+{
+    /**
+     * Safely retrieves an integer value from a ParameterBag.
+     *
+     * Uses FILTER_VALIDATE_INT with FILTER_NULL_ON_FAILURE to avoid exceptions.
+     * Returns the default value if the key doesn't exist or contains an invalid value.
+     *
+     * Examples:
+     * // Basic usage
+     * $id = ParameterBagHelper::getInt($request->query, 'id');
+     *
+     * // With default value
+     * $limit = ParameterBagHelper::getInt($request->request, 'limit', 50);
+     *
+     * // Multiple sources with fallback
+     * $id = ParameterBagHelper::getInt($request->attributes, 'id')
+     *     ?: ParameterBagHelper::getInt($request->query, 'id');
+     *
+     */
+    public static function getInt(ParameterBag $bag, string $key, int $default = 0): int
+    {
+        return $bag->filter($key, $default, \FILTER_VALIDATE_INT, \FILTER_NULL_ON_FAILURE) ?? $default;
```

Comment:
LGTM, 
just feeling Symfony is not into applying/considering the `default` option present in every `FILTER_VALIDATE_*`
https://www.php.net/manual/en/filter.constants.php#constant.filter-validate-bool
<img width="457" height="270" alt="image" src="https://github.com/user-attachments/assets/d9c90f49-9f80-4569-af9c-01e69c04a40e" />

by revolving everything around `FILTER_NULL_ON_FAILURE`
https://github.com/symfony/symfony/blob/1901cc3f1c637d2aa2fcab7fedef47a8d4e9b97a/src/Symfony/Component/HttpFoundation/ParameterBag.php#L229-L232

we could have it like

```suggestion
        return $bag->filter($key, $default, \FILTER_VALIDATE_INT, ['options' => ['default' => $default]]);
```

Reply:
Also feels like there are so many protection against an invalid value being provided, while the root problem would be that invalid values were provided. In other words, all these fallback and checks in `filter()` are getting like covered up instead of favoring prevention.